db/python/tables/family.py

```diff
@@ -29,16 +29,39 @@ async def get_projects_by_family_ids(self, family_ids: List[int]) -> Set[Project
             )
         return projects
 
-    async def get_families(self, project: int = None) -> List[Family]:
+    async def get_families(
+        self, project: int = None, participant_ids: List[int] = None
+    ) -> List[Family]:
         """Get all families for some project"""
-        _query = """\
-SELECT id, external_id, description, coded_phenotype, project
-FROM family
-WHERE project = :project"""
+        _query = """
+            SELECT id, external_id, description, coded_phenotype, project
+            FROM family
+        """
 
-        rows = await self.connection.fetch_all(
-            _query, {'project': project or self.project}
-        )
+        values: Dict[str, Any] = {'project': project or self.project}
+        where: List[str] = []
+
+        if participant_ids:
+            _query += """
+                JOIN family_participant
+                ON family.id = family_participant.family_id
+            """
+
+            placeholders = []
+            for i, pid in enumerate(participant_ids):
+                values[str(i)] = pid
+                placeholders.append(f':{i}')
+
+            pids = ', '.join(placeholders)
+            where.append(f'participant_id IN ({pids})')
+
+        if project or self.project:
+            where.append('project = :project')
```

Comment:
Kind of an unwritten rule, but if project isn't set, then you have to explicitly look up the projects that the participant / family IDs belong to and access check. So I'd advise to just always do this check and probably assert project exists.